include/seqan3/alphabet/aminoacid/aa10li.hpp

```diff
@@ -0,0 +1,220 @@
+ * H represents a group of hydrophobic residues (H,N).
+ * I represents a group of large hydrophobic residues (I,V).
+ * J represents a group of large hydrophobic residues (J,L,M).
+ * K represents long-chain positively charged residues (K,R) and the species-specific amino acid Pyrrolysine.
+ * G and P do not represent any other amino acids other than themselves.
+ *
+ * This alphabet allows to reduce the aminoacid alphabet size to 10 but is still able to recognize and represent
+ * folding of all proteins. Amino acids are grouped together based on residues.
+ *
+ * *Note:* Letters which belong in the extended alphabet will be automatically converted.
+ * Terminator characters are converted to F, because the most commonly occurring stop codon in higher eukaryotes
+ * is UGA <sup>2</sup>. This is most similar to a Tryptophan which in this alphabet
+ * gets converted to Phenylalanine. Anything unknown is converted to A.
+ *
+ * |Input Letter  |Converts to    |
+ * |--------------|---------------|
+ * |D             |B<sup>1</sup>  |
+ * |E             |B<sup>1</sup>  |
+ * |L             |J<sup>1</sup>  |
+ * |M             |J<sup>1</sup>  |
+ * |N             |H<sup>1</sup>  |
+ * |O             |K<sup>1</sup>  |
+ * |Q             |B<sup>1</sup>  |
+ * |R             |K<sup>1</sup>  |
+ * |S             |A<sup>1</sup>  |
+ * |T             |A<sup>1</sup>  |
+ * |U             |C<sup>1</sup>  |
+ * |V             |I<sup>1</sup>  |
+ * |W             |F<sup>1</sup>  |
+ * |Y             |F<sup>1</sup>  |
+ * |Z             |B<sup>1</sup>  |
+ * |X (Unknown)   |A<sup>1</sup>  |
+ * |* (Terminator)|F<sup>1,2</sup>|
+ *
+ * <sup><b>1</b></sup>T. Li, K. Fan, J. Wang, and W. Wang. Reduction of protein sequence complexity by
+ * residue grouping. Protein Eng., 16(5):323–330, May 2003.\n
+ * <sup><b>2</b></sup>Trotta, E. (2016). Selective forces and mutational biases drive stop codon usage
+ * in the human genome: a comparison with sense codon usage.
+ * BMC Genomics, 17, 366. http://doi.org/10.1186/s12864-016-2692-4
+ *
+ * \snippet test/snippet/alphabet/aminoacid/aa10li.cpp example
+ */
+class aa10li : public aminoacid_base<aa10li, 10>
+{
+private:
+    //!\brief The base class.
+    using base_t = aminoacid_base<aa10li, 10>;
+
+    //!\brief Befriend seqan3::aminoacid_base.
+    friend base_t;
+    //!\cond \brief Befriend seqan3::alphabet_base.
+    friend base_t::base_t;
+    //!\endcond
+
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    constexpr aa10li() = default;                           //!< Defaulted.
+    constexpr aa10li(aa10li const &) = default;             //!< Defaulted.
+    constexpr aa10li(aa10li &&) = default;                  //!< Defaulted.
+    constexpr aa10li & operator=(aa10li const &) = default; //!< Defaulted.
+    constexpr aa10li & operator=(aa10li &&) = default;      //!< Defaulted.
+    ~aa10li() = default;                                    //!< Defaulted.
```

Comment:
Can you add noexcept and align the identical elements? e.g.
```cpp
    constexpr aa10li()                           noexcept = default; //!< Defaulted.
    constexpr aa10li(aa10li const &)             noexcept = default; //!< Defaulted.
    constexpr aa10li(aa10li &&)                  noexcept = default; //!< Defaulted.
    constexpr aa10li & operator=(aa10li const &) noexcept = default; //!< Defaulted.
    constexpr aa10li & operator=(aa10li &&)      noexcept = default; //!< Defaulted.
    ~aa10li()                                    noexcept = default; //!< Defaulted.
```

Reply:
The base class is not noexcept - should we change this now for all amino acid alphabets and therefore also in the base class?

Reply:
Yes, that would be great! But maybe in a separate commit. Thanks!

Reply:
Then I will update all alphabets together and leave this here as it is because it does not compile when I change it only here due to the inconsistency.